app/src/main/java/com/example/georgesamuel/dubaihotels/presentation/features/details/DetailsActivity.java

```diff
@@ -0,0 +1,117 @@
+import com.example.georgesamuel.dubaihotels.entities.Hotel;
+import com.squareup.picasso.Picasso;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class DetailsActivity extends AppCompatActivity implements View.OnClickListener {
+
+    @BindView(R.id.hotel_name_details_text_view)
+    TextView hotelNameTextView;
+
+    @BindView(R.id.actual_price_text_view)
+    TextView actualPriceTextView;
+
+    @BindView(R.id.hotel_address_text_view)
+    TextView addressTextView;
+
+    @BindView(R.id.normal_price_text_view)
+    TextView normalPriceTextView;
+
+    @BindView(R.id.details_pic_image_view)
+    ImageView hotelPicImageView;
+
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @BindView(R.id.back_image_view)
+    ImageView backImageView;
+
+    @BindView(R.id.open_map_button)
+    Button openMapButton;
+
+    boolean isImageFitToScreen;
+    private Hotel hotel;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_details);
+        ButterKnife.bind(this);
+        if (getIntent() != null) {
+             hotel = (Hotel) getIntent().getSerializableExtra("Details");
+            setData(hotel);
+        }
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayShowTitleEnabled(false);
+        backImageView.setOnClickListener(this);
+        hotelPicImageView.setOnClickListener(this);
+        openMapButton.setOnClickListener(this::onClick);
+
+    }
+
+
+    private void setData(Hotel data) {
+        String imageUrl = data.getImage().get(0).getUrl();
+        String address = data.getLocation().getAddress();
+        String hotelName = data.getSummary().getHotelName();
+        Double actualPrice = data.getSummary().getLowRate();
+        Double normalPrice = data.getSummary().getHighRate();
+        Picasso.with(this).load(imageUrl).placeholder(getResources().getDrawable(R.drawable.ic_launcher_background)).into(hotelPicImageView);
+        addressTextView.setText(address);
+        hotelNameTextView.setText(hotelName);
+        normalPriceTextView.setText(String.valueOf(normalPrice));
+        actualPriceTextView.setText(String.valueOf(actualPrice));
+
+    }
+
+    @Override
+    public void onClick(View view) {
+        switch (view.getId()){
+
+            case R.id.back_image_view:
+                finish();
+                break;
+
+                case R.id.open_map_button:
+                    Intent intent=new Intent(this, MapActivity.class);
+                    if(hotel!=null){
+                    intent.putExtra("lang",hotel.getLocation().getLongitude());
+                    intent.putExtra("lat",hotel.getLocation().getLatitude());
+                    intent.putExtra("hotelName",hotel.getSummary().getHotelName());
+                    }
+
+                    startActivity(intent);
+                break;
+            case R.id.details_pic_image_view:
```

Comment:
use ButterKnife for each onClick instead of switch cases

Reply:
i replaced it